### DISCUSSION TITLE
Ability to turn off lock

### DISCUSSION THREAD

**Discussion by mstang1988** [omersiar/esp-rfid#523] [General]
I'm looking at leveraging this solution for my backyard gate.  My usecase is I'll want it locked most of the time unless I am inviting folks over and want them to come straight through the gate to the backyard.  With that in mind I would like to toggle the lock to unlocked by default.  There doesn't appear to be the capability to do this but do folks have thoughts on doing this?

Alternatively I have thought about building the project, modifying it myself etc.  What build environment are folks utilizing?  Just gpp?  I didn't see a makefile.

**Comment by mstang1988** [comment]
Ok, I found it to be built with platformio.  I am getting compile errors though on both the dev and stable versions with the Json lib.

_s<T>::type = long long int]'
.pio/libdeps/generic/ArduinoJson/src/ArduinoJson/Deserialization/../JsonVariantCasts.hpp:50:34:   required from 'ArduinoJson::Internals::JsonVariantCasts<TImpl>::operator T() const [with T = long long int; TImpl = ArduinoJson::Internals::JsonObjectSubscript<const char*>]'
src/websocket.esp:117:26:   required from here
.pio/libdeps/generic/ArduinoJson/src/ArduinoJson/JsonObject.hpp:283:46: error: no matching function for call to 'ArduinoJson::JsonVariant::as<long long int>() const'
  283 |     return it != end() ? it->value.as<TValue>()
      |_ 


_.pio/libdeps/generic/ArduinoJson/src/ArduinoJson/Deserialization/../JsonVariantCasts.hpp:50:34:   required from 'ArduinoJson::Internals::JsonVariantCasts<TImpl>::operator T() const [with T = long long int; TImpl = ArduinoJson::Internals::JsonObjectSubscript<const char*>]'
src/websocket.esp:117:26:   required from here
.pio/libdeps/generic/ArduinoJson/src/ArduinoJson/Deserialization/../JsonVariant.hpp:150:3: error: no type named 'type' in 'struct ArduinoJson::Internals::EnableIf<false, long long int>'_

**Reply by mstang1988** [reply]
I've been able to build the code now.  The issue I was seeing was related to long long on the stable branch.  I later moved to the dev branch and also did not experience the issue.

**Comment by matjack1** [comment]
hey @mstang1988 have you figured this out already? If you want the gate to be open by default, then what you expect to do in the esp-rfid? Change a state that closes the door and enables the checks? Or maybe can you think about turning the esp-rfid system on when you need the gate to be locked? If this is good enough for you, when esp-rfid boots will switch the relays in the default closed state. So maybe it could work, if turning esp-rfid on is good enough for you!

**Reply by mstang1988** [reply]
I have not (haven't had a chance to tinker this week due to work commitments).  My usecase is this:

Steady state:
- Gate Locked 7 days a week
- Gate has code that people can use to enter (lawn workers, pool people, etc).  
- Gate auto locks when closed and has notification of being opened/closed (Want to ensure it's not open when we let the dogs out)

"Party" State:
- We are in the backyard hosting an event.  People are free to come and go as they wish because we are out there.  No concerns about gate codes as it will be friends/family coming and going.  The need to enter the gate code each time during this event and handing out possible one time use codes isn't useful.  Therefore I would like to just unlock the gate (but the gate is closed).  Just the handle to open the latch and come in (vs enter key code, then latch, then open).
- 
-

**Reply by matjack1** [reply]
@mstang1988 it's been a while, but maybe you are still interested :)

Now you can have pin codes that work together with cards, so you can give pin codes to who you want, and there's also a door state sensor, to show if it's open or close.

For the party mode instead, you might manage it with a specific relay, that opens or closes the door indefinitely. Then you have a pin code (and user) that can open that lock specifically, opening the door until you trigger it again. To close you could use a third relay maybe, don't know :)
